graphql/schemabuilder/batch.go

```diff
@@ -297,7 +317,11 @@ func (funcCtx *batchFuncContext) prepareResolveArgs(sources []interface{}, args
 		idxVal := idx
 		sourceValue := reflect.ValueOf(source)
 		ptrSource := sourceValue.Kind() == reflect.Ptr
-		idxValues[idxVal] = reflect.ValueOf(idxVal)
+		if funcCtx.usesBatchIndex {
+			idxValues[idxVal] = reflect.ValueOf(batch.Index(idxVal))
+		} else {
+			idxValues[idxVal] = reflect.ValueOf(idxVal)
```

Comment:
why don't we have to do anything in the output processing?

Reply:
Because we pass the reflect.Value of the keys directly to the output parsing function